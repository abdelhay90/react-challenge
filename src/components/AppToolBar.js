import React from 'react';
import Typography from '@material-ui/core/Typography';
import { AppBar, makeStyles, Toolbar } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  appBar: {
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
  toolbar: {
    flexWrap: 'wrap',
  },
  toolbarTitle: {
    flexGrow: 1,
  },
  link: {
    margin: theme.spacing(1, 1.5),
  },
}));

export default function AppToolBar() {
  const classes = useStyles();
  return (
    <AppBar
      position='static'
      color='default'
      elevation={0}
      className={classes.appBar}
    >
      <Toolbar className={classes.toolbar}>
        <Typography
          variant='h6'
          color='inherit'
          noWrap
          className={classes.toolbarTitle}
        >
          SWVL
        </Typography>
      </Toolbar>
    </AppBar>
  );
}
