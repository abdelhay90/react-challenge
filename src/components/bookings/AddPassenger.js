import React, { useState } from 'react';
import { inject } from 'mobx-react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Fab,
  Tooltip,
} from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import PassengerForm from './PassengerForm';

const AddPassenger = inject('trip')(({ trip }) => {
  const [open, setOpen] = useState(false);
  const [addLabel, setAddLabel] = useState('Add Passenger');
  const toggleOpen = () => {
    setOpen(!open);
    if (trip.isTripMaxCapacity) {
      setAddLabel('Max Capacity');
    }
  };
  const disableClickAndEsc = true;
  return (
    <>
      <Tooltip title={addLabel} aria-label='add'>
        <Fab
          size='small'
          aria-label='Add Passenger'
          color='primary'
          onClick={toggleOpen}
          disabled={trip.isTripMaxCapacity}
        >
          <AddIcon />
        </Fab>
      </Tooltip>
      <Dialog
        fullWidth
        maxWidth='md'
        open={open}
        disableBackdropClick={disableClickAndEsc}
        disableEscapeKeyDown={disableClickAndEsc}
        onClose={toggleOpen}
      >
        <DialogTitle id='max-width-dialog-title'>
          Add Customer to Current Trip
        </DialogTitle>
        <DialogContent>
          <PassengerForm onPassengerAdded={toggleOpen} />
        </DialogContent>
        <DialogActions>
          <Button onClick={toggleOpen} color='primary'>
            Close
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
});

export default AddPassenger;
