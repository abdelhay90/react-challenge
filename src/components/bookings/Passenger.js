/**
 * Passenger card that hold the passenger identity, status, and payment method
 */
import React from 'react';
import Box from '@material-ui/core/Box';
import { Avatar, makeStyles, Typography } from '@material-ui/core';

const useStyles = makeStyles({
  passenger: {
    minWidth: 340,
    maxWidth: 340,
  },
  bigAvatar: {
    margin: 1,
    width: 60,
    height: 60,
    border: '1px solid rgba(88,88,88,0.1)',
    'box-shadow': '2px 0px rgba(88,88,88,0.3)',
  },
});

export default function Passenger({
  id,
  name,
  status,
  paymentMethod,
  imageUrl,
}) {
  const classes = useStyles();
  return (
    <Box
      id={id}
      display='flex'
      flexWrap='wrap'
      p={1}
      mx={1}
      className={classes.passenger}
    >
      <Box p={1} mx={1} alignItems='center'>
        <Avatar alt={name} src={imageUrl} className={classes.bigAvatar}>
          {imageUrl === '' ? name[0] : ''}
        </Avatar>
      </Box>
      <Box p={1} mx={1} xs={6}>
        <Typography variant='h6' gutterBottom>
          {name}
        </Typography>
        {`${status} - ${paymentMethod}`}
      </Box>
    </Box>
  );
}
