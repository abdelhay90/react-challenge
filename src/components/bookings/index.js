export Bookings from './Bookings';
export PassengerList from './PassengerList';
export Passenger from './Passenger';
export AddPassenger from './AddPassenger';
export PassengerForm from './PassengerForm';
