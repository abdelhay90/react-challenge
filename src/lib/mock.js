export const stops = [
  {
    stationId: '',
    stationName: 'Talaat Harb Axis',
    coordinates: [29.995097117000945, 31.445976804476345].reverse(),
  },
  {
    stationId: '',
    stationName: 'Omar Ibn Abdelaziz',
    coordinates: [30.012880226989733, 31.431022685763764].reverse(),
  },
  {
    stationId: '',
    stationName: '',
    coordinates: [30.016423452010145, 31.43356512343098].reverse(),
  },
  {
    stationId: '',
    stationName: 'Southern Teseen',
    coordinates: [30.016375545253084, 31.39839364118893].reverse(),
  },
  {
    stationId: '',
    stationName: 'El-Shaikh Kamel Khedr',
    coordinates: [30.040987895343214, 31.34613896298947].reverse(),
  },
  {
    stationId: '',
    stationName: 'Ahmed ElZomor',
    coordinates: [30.04505095423728, 31.34143552756268].reverse(),
  },
  {
    stationId: '',
    stationName: 'Mostafa ElNahas',
    coordinates: [30.054089469238654, 31.34217149725646].reverse(),
  },
  {
    stationId: '',
    stationName: 'Makram Ebid',
    coordinates: [30.0619822837204, 31.34523262262769].reverse(),
  },
  {
    stationId: '',
    stationName: 'Makrem Ebeid Ext',
    coordinates: [30.073225027534153, 31.343785066473743].reverse(),
  },
  {
    stationId: '',
    stationName: 'ElThawra St.',
    coordinates: [30.08201257054215, 31.343882169980134].reverse(),
  },
];

export const passengers = [
  {
    id: '5d865ee507bfe196bac94ff3',
    name: 'Bailey Walls',
    status: 'completed',
    paymentMethod: 'cash',
    imageUrl: '/static/images/leo.jpg',
  },
  {
    id: '5d865ee543724f1184914907',
    name: 'Angie Rice',
    status: 'missed',
    paymentMethod: 'cash',
    imageUrl: '/static/images/full-face-1.jpg',
  },
  {
    id: '5d865ee54ec4f96a85fa3dbc',
    name: 'Cecilia Trevino',
    status: 'canceled',
    paymentMethod: 'credit',
    imageUrl: 'static/images/full-face-2.jpg',
  },
  {
    id: '5d865ee5ad0a247f264dd75f',
    name: 'Coffey Fleming',
    status: 'completed',
    paymentMethod: 'cash',
    imageUrl: 'static/images/full-face-2.jpg',
  },
  {
    id: '5d865ee5ecfd4d136af665ba',
    name: 'Stuart Orr',
    status: 'canceled',
    paymentMethod: 'cash',
    imageUrl: '/static/images/leo.jpg',
  },
  {
    id: '5d865ee5b3da2fb19e62a897',
    name: 'Vickie Beard',
    status: 'completed',
    paymentMethod: 'cash',
    imageUrl: '/static/images/full-face-1.jpg',
  },
  {
    id: '5d865ee525b0567a26b1c2f3',
    name: 'Sandra Blackburn',
    status: 'completed',
    paymentMethod: 'cash',
    imageUrl: '/static/images/leo.jpg',
  },
  {
    id: '5d865ee5f18d6587c1cb9ed0',
    name: 'Tia Oneal',
    status: 'canceled',
    paymentMethod: 'credit',
    imageUrl: '/static/images/leo.jpg',
  },
  {
    id: '5d865ee527d6024df9300d92',
    name: 'Marylou Carr',
    status: 'missed',
    paymentMethod: 'credit',
    imageUrl: 'static/images/full-face-2.jpg',
  },
  {
    id: '5d865ee51049173f6c386500',
    name: 'Rowe Rocha',
    status: 'missed',
    paymentMethod: 'credit',
    imageUrl: '/static/images/leo.jpg',
  },
  {
    id: '5d865ee5d0d97ea49d4e5124',
    name: 'Mcconnell Jefferson',
    status: 'completed',
    paymentMethod: 'credit',
    imageUrl: '/static/images/full-face-1.jpg',
  },
  {
    id: '5d865ee580c33a45ed14c2ed',
    name: 'Cook Best',
    status: 'canceled',
    paymentMethod: 'credit',
    imageUrl: '/static/images/full-face-1.jpg',
  },
];

export const trip = {
  id: '123456789',
  distance: 47,
  rateFare: 25,
  driverInfo: {
    carModel: ' TOYOTA HiAce 2019',
    carImageUrl: '/static/images/Toyota-HiAce.jpg',
    driverName: 'Ahmed Abdelhay',
    driverImageUrl: '/static/images/driver-1.jpg',
  },
  path: stops.slice(0),
  passengers: passengers.slice(0),
};
