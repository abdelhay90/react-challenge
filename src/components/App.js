import React from 'react';
import 'mapbox-gl/dist/mapbox-gl.css';
import { Container, CssBaseline, makeStyles } from '@material-ui/core';
import AppToolBar from './core/AppToolBar';
import {
  MainMapContainer,
  TripInfoContainer,
  BookingsContainer,
} from '../containers';

const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
    ul: {
      margin: 0,
      padding: 0,
    },
    li: {
      listStyle: 'none',
    },
  },
  mainContainer: {
    padding: theme.spacing(8, 0, 6),
  },
  mapContainer: {
    'padding-top': '100px',
    'padding-right': '16px',
    'padding-left': '16px',
  },
}));

export default function App() {
  const classes = useStyles();
  return (
    <>
      <CssBaseline />
      <AppToolBar />
      <Container
        maxWidth='lg'
        component='main'
        className={`${classes.mainContainer} ${classes.mapContainer}`}
      >
        <MainMapContainer />
      </Container>
      <Container
        maxWidth='lg'
        component='main'
        className={classes.mainContainer}
      >
        <TripInfoContainer />
      </Container>
      <Container
        maxWidth='lg'
        component='main'
        className={classes.mainContainer}
      >
        <BookingsContainer />
      </Container>
    </>
  );
}
