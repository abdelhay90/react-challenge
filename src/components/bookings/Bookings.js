import React from 'react';
import { Box, Typography } from '@material-ui/core';
import Passenger from './Passenger';

function Bookings({ passengers = [] }) {
  return (
    <>
      <Box px={1} mx={1}>
        <Typography variant='h5'>Bookings</Typography>
      </Box>
      <Box display='flex' flexWrap='wrap' p={1} m={1}>
        {passengers.map(passenger => (
          <Passenger
            id={passenger.id}
            name={passenger.name}
            status={passenger.name}
            paymentMethod={passenger.name}
            imageUrl={passenger.imageUrl}
          />
        ))}
      </Box>
    </>
  );
}

export default Bookings;
