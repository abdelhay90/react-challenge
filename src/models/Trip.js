import { observable, action, computed } from 'mobx';
import groupBy from 'lodash/groupBy';
import Passenger from './Passenger';
import {
  PASSENGER_STATUS,
  PASSENGERS_MAX_CAPACITY,
  TRIP_STATUS,
} from '../lib/constants';

export default class Trip {
  id;

  @observable distance;

  @observable rateFare;

  @observable driverInfo;

  @observable path;

  @observable passengers = [];

  @observable tripStatus;

  @observable currentStop;

  constructor({
    id,
    passengers = [],
    distance,
    rateFare,
    driverInfo,
    path,
    tripStatus = TRIP_STATUS.NOT_STARTED,
  }) {
    this.id = id;
    this.passengers = passengers.map(item => new Passenger({ ...item }));
    this.distance = distance;
    this.rateFare = rateFare;
    this.driverInfo = driverInfo;
    this.path = path;
    this.currentStop = this.path.length > 0 ? this.path[0] : null;
    this.tripStatus = tripStatus;
  }

  /**
   * computed property returns the route object that help in binding to screen
   * @returns {{endPoint: string, route: *, routeStops: *, name: (string|number), startPoint: string, currentStop: *}}
   */
  @computed
  get route() {
    return {
      name:
        this.path.length !== 0
          ? `${this.path[0].stationName} - ${this.path[this.path.length - 1].stationName}`
          : 0,
      route: this.path.map(item => item.coordinates),
      routeStops: this.path,
      startPoint: this.path.length !== 0 ? this.path[0] : '',
      endPoint: this.path.length !== 0 ? this.path[this.path.length - 1] : '',
      currentStop: this.currentStop,
    };
  }

  @computed
  get stationsMap() {
    return this.path.reduce(
      (accum, value) => ({ ...accum, [value.stationId]: value }),
      {},
    );
  }

  /**
   * computed property checks if car of current trip has reached max capacity
   * @returns {boolean}
   */
  @computed
  get isTripMaxCapacity() {
    return this.passengers.length >= PASSENGERS_MAX_CAPACITY;
  }

  /**
   * computed property returns the passengers status statistics during trip
   * @returns {{x: *, y: *}[]}
   */
  @computed
  get passengersStats() {
    let allGroups = Object.keys(PASSENGER_STATUS).reduce(
      (accum, value) => ({ ...accum, [value]: [] }),
      {},
    );
    const calculatedGroups = groupBy(this.passengers, 'status');
    allGroups = { ...allGroups, ...calculatedGroups };
    return Object.keys(allGroups).map(key => ({
      x: key,
      y: allGroups[key].length,
    }));
  }

  /**
   * helper function that create the station map that can hold statistics of passengers per each station
   * @returns {*}
   */
  _getStationMap() {
    return this.path.reduce(
      (accum, value) => ({ ...accum, [value.stationId]: [] }),
      {},
    );
  }

  /**
   * combine the all map of station with filled with zero and the new station map statistics
   * @param allStations
   * @param calculated
   * @returns {{x: *, y: *}[]}
   */
  _combineStationStats(allStations, calculated) {
    const newStatsMap = { ...allStations, ...calculated };
    return Object.keys(newStatsMap).map(key => ({
      x: this.path.filter(item => item.stationId === key)[0].stationName,
      y: newStatsMap[key].length,
    }));
  }

  /**
   * computed property return the statistics of picked up passengers per each station
   * @returns {{x: *, y: *}[]}
   */
  @computed
  get stationPickupStats() {
    const allStations = this._getStationMap();
    const passengers = this.passengers.filter(
      passenger =>
        passenger.status !== PASSENGER_STATUS.MISSED &&
        passenger.status !== PASSENGER_STATUS.CANCELLED,
    );
    const stationGroups = groupBy(passengers, 'pickupStationId');
    return this._combineStationStats(allStations, stationGroups);
  }

  /**
   * computed property return the statistics of checked out passengers per each station
   * @returns {{x: *, y: *}[]}
   */
  @computed
  get stationCheckoutStats() {
    const allStations = this._getStationMap();
    const passengers = this.passengers.filter(
      passenger => passenger.status === PASSENGER_STATUS.COMPLETED,
    );
    const stationGroups = groupBy(passengers, 'checkoutStationId');
    return this._combineStationStats(allStations, stationGroups);
  }

  /**
   * action to add new passenger to passengers list
   * @param passenger
   */
  @action.bound
  addPassenger(passenger) {
    if (this.passengers.length < 12)
      this.passengers.push(new Passenger({ ...passenger }));
  }

  /**
   * action to remove existing passenger from passengers list
   * @param item
   */
  @action.bound
  removePassenger(item) {
    this.passengers = this.passengers.filter(i => i !== item);
  }

  /**
   * action to set start trip status
   */
  @action.bound
  startTrip() {
    this.tripStatus = TRIP_STATUS.STARTED;
  }

  /**
   * action to set stop trip status
   */
  @action.bound
  stopTrip() {
    this.tripStatus = TRIP_STATUS.FINISHED;
  }
}
