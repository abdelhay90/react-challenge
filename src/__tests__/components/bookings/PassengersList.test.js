import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import Trip from '../../../models/Trip';
import { trip } from '../../../lib/mock';
import { PassengerList } from '../../../components/bookings';

describe('Passenger List Component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    const currentTrip = new Trip({ ...trip });
    ReactDOM.render(
      <Provider trip={currentTrip}>
        <PassengerList passengers={currentTrip.passengers} />
      </Provider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should match snapshot', () => {
    const currentTrip = new Trip({ ...trip });
    const tree = renderer
      .create(
        <Provider trip={currentTrip}>
          <PassengerList passengers={currentTrip.passengers} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show more or less', () => {
    const currentTrip = new Trip({ ...trip });
    const wrapper = mount(
      <Provider trip={currentTrip}>
        <PassengerList passengers={currentTrip.passengers} />
      </Provider>,
    );
    expect(
      wrapper
        .find('button')
        .last()
        .getDOMNode().textContent,
    ).toEqual('View More');
    const evt = new window.Event('click', { bubbles: true });
    evt.simulated = true;

    wrapper
      .find('button')
      .last()
      .simulate('click');
    expect(
      wrapper
        .find('button')
        .last()
        .getDOMNode().textContent,
    ).toEqual('View Less');
  });
});
