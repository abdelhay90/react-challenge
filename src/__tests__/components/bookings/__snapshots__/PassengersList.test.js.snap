// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passenger List Component should match snapshot 1`] = `
Array [
  <div
    className="MuiCollapse-container"
    style={
      Object {
        "minHeight": "300px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          className="MuiBox-root MuiBox-root-102"
        >
          <div
            className="MuiBox-root MuiBox-root-105 makeStyles-passenger-103"
            id="5d865ee507bfe196bac94ff3"
          >
            <div
              className="MuiBox-root MuiBox-root-106"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Bailey Walls"
                  className="MuiAvatar-img"
                  src="/static/images/leo.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-110"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Bailey Walls
              </h6>
              COMPLETED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-141 makeStyles-passenger-103"
            id="5d865ee543724f1184914907"
          >
            <div
              className="MuiBox-root MuiBox-root-142"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Angie Rice"
                  className="MuiAvatar-img"
                  src="/static/images/full-face-1.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-143"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Angie Rice
              </h6>
              MISSED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-144 makeStyles-passenger-103"
            id="5d865ee54ec4f96a85fa3dbc"
          >
            <div
              className="MuiBox-root MuiBox-root-145"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Cecilia Trevino"
                  className="MuiAvatar-img"
                  src="static/images/full-face-2.png"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-146"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Cecilia Trevino
              </h6>
              CANCELLED - credit
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-147 makeStyles-passenger-103"
            id="5d865ee5ad0a247f264dd75f"
          >
            <div
              className="MuiBox-root MuiBox-root-148"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Coffey Fleming"
                  className="MuiAvatar-img"
                  src="static/images/full-face-2.png"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-149"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Coffey Fleming
              </h6>
              COMPLETED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-150 makeStyles-passenger-103"
            id="5d865ee5ecfd4d136af665ba"
          >
            <div
              className="MuiBox-root MuiBox-root-151"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Stuart Orr"
                  className="MuiAvatar-img"
                  src="/static/images/leo.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-152"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Stuart Orr
              </h6>
              CANCELLED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-153 makeStyles-passenger-103"
            id="5d865ee5b3da2fb19e62a897"
          >
            <div
              className="MuiBox-root MuiBox-root-154"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Vickie Beard"
                  className="MuiAvatar-img"
                  src="/static/images/full-face-1.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-155"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Vickie Beard
              </h6>
              COMPLETED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-156 makeStyles-passenger-103"
            id="5d865ee525b0567a26b1c2f3"
          >
            <div
              className="MuiBox-root MuiBox-root-157"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Sandra Blackburn"
                  className="MuiAvatar-img"
                  src="/static/images/leo.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-158"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Sandra Blackburn
              </h6>
              COMPLETED - CASH
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-159 makeStyles-passenger-103"
            id="5d865ee5f18d6587c1cb9ed0"
          >
            <div
              className="MuiBox-root MuiBox-root-160"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Tia Oneal"
                  className="MuiAvatar-img"
                  src="/static/images/leo.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-161"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Tia Oneal
              </h6>
              CANCELLED - CREDIT
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-162 makeStyles-passenger-103"
            id="5d865ee527d6024df9300d92"
          >
            <div
              className="MuiBox-root MuiBox-root-163"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Marylou Carr"
                  className="MuiAvatar-img"
                  src="static/images/full-face-2.png"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-164"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Marylou Carr
              </h6>
              MISSED - CREDIT
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-165 makeStyles-passenger-103"
            id="5d865ee51049173f6c386500"
          >
            <div
              className="MuiBox-root MuiBox-root-166"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Rowe Rocha"
                  className="MuiAvatar-img"
                  src="/static/images/leo.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-167"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Rowe Rocha
              </h6>
              MISSED - CREDIT
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-168 makeStyles-passenger-103"
            id="5d865ee5d0d97ea49d4e5124"
          >
            <div
              className="MuiBox-root MuiBox-root-169"
            >
              <div
                className="MuiAvatar-root makeStyles-bigAvatar-104"
              >
                <img
                  alt="Mcconnell Jefferson"
                  className="MuiAvatar-img"
                  src="/static/images/full-face-1.jpg"
                />
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-170"
              xs={6}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Mcconnell Jefferson
              </h6>
              COMPLETED - CREDIT
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "left": "50%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      View More
    </span>
  </button>,
]
`;
